### DISCUSSION TITLE
How to use a temporary e-mail with cypress

### DISCUSSION THREAD

**Discussion by PauloRFilho** [cypress-io/cypress#27074] [Questions and Help]
I'm looking for a way to use a temporary email so that my system can register a user. I even tried https://10minutemail.com/ but without success. I would like other suggestions and how I could apply this.

**Comment by portyu9** [comment]
Hi @PauloRFilho 

I would avoid automating the **10MinuteMail UI itself** from Cypress.

Cypress recommends handling email programmatically instead of visiting and interacting with a third-party mailbox UI. A disposable-email website adds another application's UI, timing, cookies/session lifetime, and availability to your test.

For registration/verification flows, a more reliable setup is one of these approaches:

- use a local test SMTP server/mail catcher in your test environment
- use a dedicated test inbox provider that exposes an API
- expose a test-only backend endpoint that lets the test retrieve the queued email

Your Cypress test can submit the registration form normally, then use `cy.request()` or a custom `cy.task()` to query the mailbox/API and retrieve the verification email. You can extract the verification URL from the message and then visit/request that URL.

That keeps the test focused on **your application** rather than testing whether 10MinuteMail's UI happens to work.

If you need disposable addresses specifically, choose a service intended for automated testing and with an API rather than controlling its public web inbox.

Please mark this answer as accepted if it helped, thank you.
